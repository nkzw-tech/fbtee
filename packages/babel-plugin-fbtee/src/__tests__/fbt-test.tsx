import { describe, expect, it } from '@jest/globals';
import {
  jsCodeFbtCallSerializer,
  snapshotTransform,
  withFbtImportStatement,
} from './FbtTestUtil.tsx';

expect.addSnapshotSerializer(jsCodeFbtCallSerializer);

describe('fbt() API: ', () => {
  describe('using FBT subject', () => {
    it('should accept "subject" as a parameter', () => {
      expect(
        snapshotTransform(
          withFbtImportStatement(`fbt("Foo", "Bar", {subject: foo});`),
        ),
      ).toMatchSnapshot();
    });
  });

  describe('using FBT subject with string templates', () => {
    it('should accept "subject" as a parameter', () => {
      expect(
        snapshotTransform(
          withFbtImportStatement('fbt(`Foo`, "Bar", {subject: foo});'),
        ),
      ).toMatchSnapshot();
    });
  });
});

describe('Test double-lined params', () => {
  it('should remove the new line for param names that are two lines', () => {
    expect(
      snapshotTransform(
        withFbtImportStatement(
          `<fbt desc="d">
          <fbt:param
            name="two
                  lines">
            <b>
              <fbt desc="test">simple</fbt>
            </b>
          </fbt:param>
          test
        </fbt>`,
        ),
      ),
    ).toMatchSnapshot();
  });
});

const describeFbtBindingTestCase = (
  requireStatement: string,
) => `${requireStatement};
      fbt("Foo", "Bar");`;

describe('fbt variable binding detection', () => {
  it(`should handle commonJS require()`, () => {
    expect(
      snapshotTransform(
        describeFbtBindingTestCase(`const fbt = require('fbtee')`),
      ),
    ).toMatchSnapshot();
  });

  describe('using ES6', () => {
    it(`should handle fbt default export`, () => {
      expect(
        snapshotTransform(
          describeFbtBindingTestCase(`import fbt from 'fbtee'`),
        ),
      ).toMatchSnapshot();
    });
    it(`should handle the named fbt export`, () => {
      expect(
        snapshotTransform(
          describeFbtBindingTestCase(`import {fbt} from 'fbtee'`),
        ),
      ).toMatchSnapshot();
    });
  });
});

test('fragments inside of <fbt:param>', () => {
  expect(
    snapshotTransform(
      withFbtImportStatement(
        `<fbt desc="d">
          <fbt:param name="param">
            <>
              <b>Test</b>
              <fbt desc="test">simple</fbt>
            </>
          </fbt:param>
          test
        </fbt>`,
      ),
    ),
  ).toMatchSnapshot();
});
