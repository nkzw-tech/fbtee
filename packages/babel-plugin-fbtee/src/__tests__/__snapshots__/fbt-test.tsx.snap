// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test double-lined params should remove the new line for param names that are two lines 1`] = `
import { fbt } from "fbtee";
fbt._(
  "{two lines} test",
  [
    fbt._param(
      "two lines",
      /*#__PURE__*/ React.createElement(
        "b",
        null,
        fbt._("simple", null, {
          hk: "2pjKFw",
        })
      )
    ),
  ],
  {
    hk: "2xRGl8",
  }
);

`;

exports[`fbt variable binding detection should handle commonJS require() 1`] = `
const fbt = require("fbtee");
fbt._("Foo", null, {
  hk: "3ktBJ2",
});

`;

exports[`fbt variable binding detection using ES6 should handle fbt default export 1`] = `
import fbt from "fbtee";
fbt._("Foo", null, {
  hk: "3ktBJ2",
});

`;

exports[`fbt variable binding detection using ES6 should handle the named fbt export 1`] = `
import { fbt } from "fbtee";
fbt._("Foo", null, {
  hk: "3ktBJ2",
});

`;

exports[`fbt() API:  using FBT subject should accept "subject" as a parameter 1`] = `
import { fbt } from "fbtee";
fbt._(
  {
    "*": "Foo",
  },
  [fbt._subject(foo)],
  {
    hk: "7I4k2",
  }
);

`;

exports[`fbt() API:  using FBT subject with string templates should accept "subject" as a parameter 1`] = `
import { fbt } from "fbtee";
fbt._(
  {
    "*": "Foo",
  },
  [fbt._subject(foo)],
  {
    hk: "7I4k2",
  }
);

`;

exports[`fragments inside of <fbt:param> 1`] = `
import { fbt } from "fbtee";
fbt._(
  "{param} test",
  [
    fbt._param(
      "param",
      /*#__PURE__*/ React.createElement(
        React.Fragment,
        null,
        /*#__PURE__*/ React.createElement("b", null, "Test"),
        fbt._("simple", null, {
          hk: "2pjKFw",
        })
      )
    ),
  ],
  {
    hk: "154Qct",
  }
);

`;
